frontend/src/ItemModule/CreateNewItem/components/CreateNewItem.vue

```diff
@@ -37,54 +43,29 @@ const selectedRarityName = form.rarityId
   ? raritiesHash.value?.[form.rarityId]?.rarity_name ?? ''
   : '';
 
-onMounted(() => {
-  fetchGames();
+onMounted(async () => {
+  const fetchedGames = await fetchGames();
+
+  games.value = fetchedGames;
 });
 
-const fetchGames = () => {
-  axios
-    .get('/game/games/')
-    .then((response) => {
-      games.value = response.data;
-    })
-    .catch((error) => {});
-};
+watch(gameId, async (gameId) => {
+  if (!gameId) return;
 
-const fetchRarities = () => {
-  if (form.gameId === null) return;
-
-  axios
-    .get(`/game/rarities/`, { params: { game_id: form.gameId } })
-    .then((response) => {
-      const fetchedRarities = response.data;
-      rarities.value = fetchedRarities;
-      raritiesHash.value = keyBy(fetchedRarities, 'id');
-    })
-    .catch((error) => {
-      console.error(error);
-    });
-};
+  const fetchedRarities = await fetchRarities({ gameId });
 
-const onSubmit = () => {
-  if (!form.chance || !form.itemName || !form.rarityId || !form.selectedImage) return;
+  rarities.value = fetchedRarities;
 
-  const formData = new FormData();
+  raritiesHash.value = keyBy(fetchedRarities, 'id');
+});
 
-  formData.append('rarity_id', form.rarityId.toString());
-  formData.append('item_name', form.itemName);
-  formData.append('image', form.selectedImage);
-  formData.append('chance', Number(form.chance).toString());
+const onCreateNewItem = () => {
+  if (!form.chance || !form.itemName || !form.rarityId || !form.selectedImage) return;
```

Comment:
Should we perform the null check inside the function instead?

Reply:
yea, I think we should do the check here. Doing it in the api function makes the return type inconsistent.